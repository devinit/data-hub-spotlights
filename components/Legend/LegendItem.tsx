import React, { FunctionComponent } from 'react';
import styled from 'styled-components';

export interface LegendItemProps {
  className?: string;
  bgColor?: string;
  textColor?: string;
}

export const LegendItem: FunctionComponent<LegendItemProps> = ({ children, className }) =>
  <span className={ className } data-testid="spotlight-legend-item">
    { children }
  </span>;

export const StyledLegendItem = styled(LegendItem)`
  && {
    background-color: ${props => props.bgColor || 'inherit'}
    color: ${props => props.textColor || 'inherit'}
  }
`;
