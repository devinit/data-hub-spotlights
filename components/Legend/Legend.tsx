import React, { Children, FunctionComponent, isValidElement } from 'react';
import { LegendItem, StyledLegendItem } from './LegendItem';

const Legend: FunctionComponent = ({ children }) => {
  return (
    <div className="spotlight-legend" data-testid="spotlight-legend">
      {
        Children.map(children, child => {
          if (isValidElement(child) && (child.type === LegendItem || child.type === StyledLegendItem)) {
            return child;
          }
        })
      }
    </div>
  );
};

export { Legend };
