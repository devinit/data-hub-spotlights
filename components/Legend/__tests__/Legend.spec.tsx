/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { render } from '@testing-library/react';
import * as TestRenderer from 'react-test-renderer';
import { Legend } from '../Legend';
import { LegendItem } from '../LegendItem';

describe('Legend', () => {
  test('renders with the class spotlight-legend', () => {
    const { container } = render(<Legend />);

    expect(container.firstChild).toHaveClass('spotlight-legend');
  });

  test('renders only LegendItem children', () => {
    const renderer = TestRenderer.create(
      <Legend>
        <LegendItem />
        Non Legend Item
        <LegendItem />
        <span>Testing</span>
      </Legend>
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
