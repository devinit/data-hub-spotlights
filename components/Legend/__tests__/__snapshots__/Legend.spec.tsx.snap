// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legend renders only LegendItem children 1`] = `
<div
  className="spotlight-legend"
  data-testid="spotlight-legend"
>
  <span
    data-testid="spotlight-legend-item"
  >
    <style
      jsx={true}
    >
      
      span {
        background-color: #D1CBCF;
        color: inherit;
      }
    
    </style>
  </span>
  <span
    data-testid="spotlight-legend-item"
  >
    <style
      jsx={true}
    >
      
      span {
        background-color: #D1CBCF;
        color: inherit;
      }
    
    </style>
  </span>
</div>
`;
